### ISSUE TITLE
Change Usernames

### ISSUE THREAD

**Issue by schwindelbub** [piwik/piwik#7864]
Hi,

it would be nice to have as an admin the ability to change the usernames of the piwik user.

My suggestion on how to proceed :
First step: Add a user ident to the "piwik_user" table.
Second step: Change all user relations in the datebase in the upgrade process to this new ident.
Third step: Make the username editable through the piwk backend for all user with the "superuser_access" flag.

Thanks for the great software and best regards
Schwindelbub

**Comment by rysiekpl** [comment]
Any chance of this happening?

**Comment by mattab** [comment]
not anytime soon

**Comment by Chardonneaur** [comment]
Hey, i have a client today who has this specific need.

**Comment by ldesgrange** [comment]
Yeah, definitely need to have a primary key that is not the `login` field but also that is not a text field. And add foreign keys where they need to be. The database schema seems too fragile here.
